poi-radio/src/lib.rs

```diff
@@ -155,12 +155,37 @@ impl RadioPayloadMessage {
 
     /// Check duplicated fields: payload message has duplicated fields with GraphcastMessage, the values must be the same
     pub fn valid_outer(&self, outer: &GraphcastMessage<Self>) -> Result<&Self, BuildMessageError> {
-        if self.nonce == outer.nonce
-            && self.graph_account == outer.graph_account
-            && self.identifier == outer.identifier
-        {
+        let mut error_messages = Vec::new();
+
+        if self.nonce != outer.nonce {
+            error_messages.push(format!(
+                "Nonce mismatch: self={}, outer={}",
+                self.nonce, outer.nonce
+            ));
+        }
+
+        if self.graph_account.to_ascii_lowercase() != outer.graph_account.to_ascii_lowercase() {
+            error_messages.push(format!(
+                "Graph account mismatch: self={}, outer={}",
+                self.graph_account, outer.graph_account
+            ));
+        }
+
+        if self.identifier.to_ascii_lowercase() != outer.identifier.to_ascii_lowercase() {
+            error_messages.push(format!(
+                "Identifier mismatch: self={}, outer={}",
+                self.identifier, outer.identifier
+            ));
+        }
+
+        if error_messages.is_empty() {
             Ok(self)
         } else {
+            let error_message = error_messages.join("; ");
```

Comment:
I think this might be a bit too verbose...? you can use the new logging format, include two level of messages and the checked results in the error
```
            Err(BuildMessageError::InvalidFields(anyhow::anyhow!(
                "Radio message wrapped by inconsistent GraphcastMessage: {:#?} <- {:#?}\nnonce check: {:#?}\naccount check: {:#?}\nidentifier check: {:#?}",
                &self,
                &outer,
                self.nonce == outer.nonce,
                self.graph_account == outer.graph_account,
                self.identifier == outer.identifier,
            )))
```